### ISSUE TITLE
Anyone else Banned?

### ISSUE THREAD

**Issue by Hmmmm69** [ExtraConcentratedJuice/pokecord-catcher#186]
I have had no problems with this for about 3 months. And this last month I have had 3 accounts banned. Anyone else having this problem? Or does anyone know what changed this last month? The accounts were discord banned.

**Comment by BeemoBeemo** [comment]
Yes, it seems discord has changed their protocol. I'm not sure how they do it, but it seems we need to find out what makes their anti-spam tick.. 

I've tried a few things, but all ended up in disabled accounts. If anyone finds a way to prevent them from being disabled, please let me know.

**Comment by ajmeese7** [comment]
Check my spambot repo [here](https://github.com/ajmeese7/spambot). I've been using it for a long time, as have several other people, and none of us have been banned. Discord has always had a hostile policy towards bots, but it seems they are finally starting to crack down on it. 

In full honesty, I've never used this bot, so I'm not sure how it works, but if it's spamming messages like my bot you can either change the time gap between messages or add in some random elements so it isn't the same thing every time. If you guys need help with this, I can look into it when I have some free time.

**Comment by BeemoBeemo** [comment]
@ajmeese7 
Hey, thanks for the help! I'm not a programmer or expert at bots at all, but I've built some experience on this bot.

In a short summary, the purpose is to spam in order to spawn Pokemon. It's quite similar to your bot, except it doesn't remove the message after, and uses a short list of sentences and words to randomly send from every second or so. 

"SpamChannelID": 01010100,
	"SpammerMinDelay": 4000,
	"SpammerMaxDelay": 5500,
	"SpammerMessages": ["david has 10 bowling balls", "my name jeff", etc.] I put about 15 different messages.

In order to maximize spawns, one dc account uses 3 separate spambots over 3 different private servers, delay has as you can see min/max and I've customized it to prevent rate limitations. 

So it does have random delays in messages, and sends randomly from a list of about 15 messages.

A separate account is used as a catcher, put in all those servers at the same time, and collects the pokemon the spammer spawns.

So the difference of why my bots get banned and not yours might be:

-The fact that your messages remove itself;
-The sheer amount of spamming in 4 different servers at once (though still not enough to be rate limited by the chillzone);
-the repeating 15 messages (they are sent randomly), i believe yours counts upwards making it endlessly different.

I've tried creating fresh servers in case the old ones were flagged or anything but this didn't matter. They also don't get banned on the spot but a few hours later after use. Also, it started to happen from the 19th of Juli, before there were no issues.

Please let me know what you think, hope these findings might be helpful to you as well, thanks again!

**Comment by ajmeese7** [comment]
@BeemoBeemo so is the Pokemon spawning just done based on sheer number of messages sent, like the XP on Discord bots like Tatsumaki and Mee6? Because if so, then I would just recommend trying the for loop method for sending endlessly different messages and maybe limiting the number of servers you are spamming in at once. If messages are sent in multiple servers very close to each other, that might be an issue. Removing the messages might also help, especially for your purpose because I send another message in the chat that deletes both the messages so two times the Pokemon. If you wanted to get really fancy with it, you could have a dictionary word file and pull by index from a randomly generated number to string a sentence together, but that seems a little out of scope.

**Comment by BeemoBeemo** [comment]
@ajmeese7 messages can "randomly" trigger a pokemon to spawn. The number of messages doesn't matter afaik. 

I'm currently testing different types of spamming (less servers, longer delay time, etc.) to see what gets disabled and what doesn't. Once I'm done I'll try your method of removing messages. 

The dictionary idea sounds great, but goes far beyond what I'm capable of. If there is a long list of sentences in the following format: "sentence",    on the internet, i could just copy and make the list of random words very long. 

However i think Discord is really cracking down on pokecord related bots in general. I've spoken with an employee, and they said were very aware of all the pokecord catch bots spamming discord's API.

I will never use these bots to spam other people or annoy anyone with it, everything is done in private servers. It might have some effect on the discord servers, so maybe that's why it's a problem for them.
